Add and pass tests

pietgeursen opened the pull request:
@ahdinosaur Review thiiiiiissssss.

ahdinosaur commented on test/dispatch.js:
```diff
@@ -0,0 +1,23 @@
+    },
+    update: function (model, action) {
+      t.deepEqual(action, expectedAction, 'action passed to update is the action passed to dispatch')
+      t.end()
```
can we also listen to the `actions` stream returned from `start(app)` for the action?

ahdinosaur commented on test/dispatch.js:
```diff
@@ -0,0 +1,23 @@
+      return {model: model}
+    },
+    view: function (model, dispatch) {
+      dispatch(expectedAction)
```
i wonder if we should also test for dispatching sometime later, even just after `process.nextTick`.

ahdinosaur commented on test/dispatch.js:
```diff
@@ -0,0 +1,23 @@
+      return {model: initialModel}
+    },
+    update: function (model, action) {
+      t.deepEqual(action, expectedAction, 'action passed to update is the action passed to dispatch')
```
for these checks you can use `t.equal`, they should be `===` since they should be the same object.

ahdinosaur commented on test/run.js:
```diff
@@ -0,0 +1,78 @@
+    }
+  }
+  inu.start(app)
+})
```
also maybe test for listening to actions stream in `run`?

ahdinosaur commented on test/update.js:
```diff
@@ -0,0 +1,29 @@
+
+test('models returned by update are emitted by the model stream', function (t) {
+  var initialModel = {initial: true}
+  var expectedModel = {type: 'EXPECTED_MODEL'}
```
this looks like an action or effect :wink: (not that it really matters...)

ahdinosaur commented on test/update.js:
```diff
@@ -0,0 +1,29 @@
+  var streams = inu.start(app)
+
+  pull(streams.models(), pull.take(2), pull.collect(function (err, models) {
+    t.false(err)
```
`t.error` ?

ahdinosaur commented:
@pietgeursen hey, thanks for the sweet tests. :tada: made some comments.

some thoughts:

- maybe rename `initialRender` test to `view`, to match your naming here.
- i like how you used the returned models stream in your model test, maybe we should be checking these streams are correct in more of these tests?
- missing tests to (i think)  get 100% code coverage (not that 100% means we've deeply tested everything, just touched a bit on everything):
  - add test to listen to actions stream within `run`
  - add test to call `stop` returned by `start` and expect all streams to disconnect and nothing to happen anymore.

pietgeursen commented:
@ahdinosaur Ok, I've added more tests and added your suggestions. There's still more to go with checking all the streams are emitting but this seems like a good start.

There are two skipped tests in the init test that I think should pass. Should they?

ahdinosaur commented:
@pietgeursen churr purrr :smile_cat:
